cmd/convox/env.go

```diff
@@ -121,12 +129,49 @@ func cmdEnvGet(c *cli.Context) error {
 	return nil
 }
 
+// Validate environment variable keypair format; trim spaces and surrounding single quotes.
+func cleanEnvPair(value string) (string, error) {
+	// Don't let CLI swallow panics. (ref. urfave/cli/issues/451)
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("ERROR: ", r, string(debug.Stack()))
+		}
+	}()
+
+	parts := strings.SplitN(value, "=", 2)
+	if len(parts) != 2 {
+		return "", errors.New("Environment variables should be defined in key=value format. You specified: " + value)
+	}
+
+	if key := strings.TrimSpace(parts[0]); key != "" {
+		val := parts[1]
+		if len(val) == 0 {
+			return "", errors.New("Can't set " + key + " to an empty value; try `convox env unset`.")
+		}
+
+		// heroku env -s adds leading and trailing single quotes to val. Strip.
+		if string(val[0]) == "'" && string(val[len(val)-1]) == "'" {
```

Comment:
Seems like we're getting into regex territory for the parsing here. Can all of these be done with the right regex?